### PR TITLE
issue-dot-field-variables-showing-duplicated-keys-V4 - Tests migrated to HostedComponent

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.spec.ts [hunk 1]
```diff
@@ -98,125 +108,124 @@ describe('DotContentTypeFieldsVariablesTableRowComponent', () => {
...
+        expect(inputs[0].nativeElement.placeholder).toContain('Enter Value');
+        expect(btns[0].nativeElement.innerText).toContain('delete_outline');
+        expect(btns[1].nativeElement.innerText).toContain('edit');
+        expect(comp.saveDisabled).toBe(false);
     });
 
-    it('should focus on "Key" input when an empty variable is added', fakeAsync(() => {
-        comp.fieldVariable = { key: '', value: '' };
-        fixture.detectChanges();
+    xit('should focus on "Key" input when an empty variable is added', () => {
```
Comment (freddyucv) [inline src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.spec.ts]: why is this one skipping?

File: src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.spec.ts [hunk 2]
```diff
@@ -98,125 +108,124 @@ describe('DotContentTypeFieldsVariablesTableRowComponent', () => {
...
+        hostComponentfixture.detectChanges();
         spyOn(comp.keyCell.nativeElement, 'click');
-        tick();
-        expect(comp.saveDisabled).toBe(false);
-        expect(comp.keyCell.nativeElement.click).toHaveBeenCalled();
-    }));
+        setTimeout(() => {
+            expect(comp.saveDisabled).toBe(false);
+            expect(comp.keyCell.nativeElement.click).toHaveBeenCalled();
+        }, 0);
```
Comment (freddyucv) [inline src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.spec.ts]: why do we need this timeout?

Comment (alfredo-dotcms) [reply src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.spec.ts]: https://github.com/dotCMS/core-web/blob/ea85dfd84ae398c829a0260972ff78480e1afddf/src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.ts#L70

Comment (fmontes) [reply src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.spec.ts]: Did you try:

```
hostComponentfixture.whenStable().then(() => {
    expect(comp.saveDisabled).toBe(true);
    expect(comp.keyCell.nativeElement.click).toHaveBeenCalled();
    done();
});
```

???

Comment (alfredo-dotcms) [reply src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.spec.ts]: yes

**Pull request merged**